### DISCUSSION TITLE
NGA 属实卧虎藏龙

### DISCUSSION THREAD

**Discussion by mark9804** [Arondight/Adachi-BOT#947] [General]
最近开了一个 [碧蓝档案的剧情查看工具](https://github.com/ba-archive/blue-archive-story-viewer) 的坑，工作量比较大想在 NGA 找有意向的合作者一起来写，

发帖不到一天已经有了一个4前端2后端2翻译1全栈的团队，直接从纯前端项目变成了前后端分离……

**Comment by BTMuli** [comment]
好羡慕，我这写个邮件安卓应用都感觉很吃力，smtp跟pop3协议本来要实现的，现在打算直接调包了ಥ_ಥ

**Reply by mark9804** [reply]
是你毕设还是什么的吗，我感觉现在邮件app都卷差不多了

**Reply by BTMuli** [reply]
没，就计网实验，毕设还有一年呢π_π

**Reply by mark9804** [reply]
你们大学计网实验整这么大的吗……开了眼了

**Reply by BTMuli** [reply]
不好说，我们有门课团队编程做的健康码小程序（*゜ー゜*）

**Comment by Arondight** [comment]
1. NGA 真牛
2. 老大哥玩的游戏真多

**Reply by mark9804** [reply]
我最多主游+副游两个，现在原因为用电脑还是有点坐牢所以在学校群里找了个黑奴，自己躺床上玩平板

**Reply by BTMuli** [reply]
我自己是原神+Pjsk（Project Sekai）+Arcaea，都是单机游戏（

**Reply by mark9804** [reply]
音游佬是吧，我曾经是deemo + cytus，线下偶尔打两把jubeat和太鼓

~~Arcaea怎么能算单机游戏~~

**Reply by BTMuli** [reply]
只会下落式，cytus对我来说还是太早了，pjsk刚玩不到两个月，Arcaea这两天刚下回来

~~Arcaea云端数据存了418天~~

**Reply by Arondight** [reply]
哎，我自从回老家 wfh 之后，越来越没空玩游戏了，有时间基本上就是跟同学吃喝玩乐，前一阵为了玩方便报了个驾照，开春买个车，现在技术也不常看了，感觉人要废掉了

**Reply by Arondight** [reply]
前几天300块钱找了个网友，把须弥的主线做了，锚点都开了，传说任务清了，森林书沙漠书做完了，不知道给多了少了，我找代练总害怕开科技封了号

**Reply by BTMuli** [reply]
富佬有钱，我也想挣（）

**Reply by Arondight** [reply]
平头百姓，一辆车四五万就行，新车的话七万就挺好了，周末出去玩玩，吃个饭一个人二三十，人均一百玩一天，晚上约个公园打打球不花钱，正常上个班，一个月五六千就够开销了

**Reply by BTMuli** [reply]
我问过代肝的群友了，把须弥的主线做了，锚点都开了，传说任务清了，森林书沙漠书做完了，700算少了，你血赚（）

**Reply by BTMuli** [reply]
所以除了开宝箱收集圣章石这种可以考虑一下吗？（富佬饿饿饭饭）

**Reply by Arondight** [reply]
那我的确血赚，网友还给我做了六个传说任务，沙漠书后续也给我做了

**Reply by BTMuli** [reply]
你这网友不会是未成年吧，又玩到了游戏又有钱赚(恶意揣测

**Reply by mark9804** [reply]
吃喝玩乐不是挺好的，我不吃喝玩乐是因为找不到人吃喝玩乐，，，

代肝价格虚高，找大学生苦力又便宜又放心

-- 发送自路灯杆顶

**Reply by Arondight** [reply]
> 你这网友不会是未成年吧，又玩到了游戏又有钱赚(恶意揣测

上班族，周末两天干完了

**Reply by mark9804** [reply]
我找群友代两个月，除了全宝箱之外其他全包那种，一个月100，我最后给了个328

去年我考研的时候问过明日方舟代肝的价格，对方开口一月500不包活动，我连夜又抓了个群友

**Comment by mark9804** [comment]
问一下老大哥，你们新项目团队是怎么建立代码规范的（代码规范长什么样），前端团队快要动手了还没个规范

**Reply by BTMuli** [reply]
Google代码规范？不过这些不是通过Eslint Prettier Stylelint 之类的来约束吗

**Reply by mark9804** [reply]
我不清楚，但是 linter 是用来“约束”的，在约束之前应该有一个规范，比如说我们项目禁止 any 战士出没，linter 负责执行禁止 any 战士的规范，就是这个规范我不清楚是什么形式应该怎么定

**Reply by Arondight** [reply]
没有规范，也没有 lint ，代码排版不怎么关注，语法上的东西大家都有最基本的共识，其余的部分也没啥要求，例如文件内函数要不要修饰静态，编译不出问题就没人管

**Reply by Arondight** [reply]
新项目一般是

1. 确定要干什么
2. 确定用到的语言框架

好了开始写，能够在可以编译的情况下完成功能即可。

例如你们确定了 react/ts ，如果我在做的话我就不关心具体的这些，先把功能实现，不喜欢 any 战士我自己不会做，但我不会管别人的代码。当然这只是我的想法，具体怎么样看人。

但是还是劝一句开源的东西在大家热情消退前赶紧动手把基础功能做个七七八八，代码难看就难看有了架子之后就有继续开发完善的可能，各种东西把人的热情都拖或者消耗没了，后续就不好搞了

**Reply by mark9804** [reply]
几个前后端那边好像确实有共识，唯一一个用 options API 的给其他 composition API 的按下去了 （用 options 的理由是 composition API `defineProps` 导入会报错）

- 函数命名 : 大驼峰
- 私有变量 : 小驼峰
- 公有变量 : 小驼峰
- 常量    : 全大写 _ 分隔
- 函数    : function
- 组件名称 : 大驼峰
- 模板    : 大驼峰

**Reply by BTMuli** [reply]
我刚开始学vue觉得组合式Api挺不错的，跟py之类的语法结构差不多；但因为同组的打算学选项式，我就写了选项式；现在看来组合式好像是更优解？

**Reply by mark9804** [reply]
你得问问人为什么选的选项式，这个时间节点学选项式有点逆天而行的意思在了…… 我们选择选项式的前端的理由是组合式 API 不能 import type，被摁下去的原因是因为其他人都用的 interface 没这问题（预计 3.3 修复），实际上要是没这个问题的话大家都喜欢组合式

我个人之前是选项式的，现在也切换到组合式了，确实写着爽

组合式的一个不好的地方是如果个人编程习惯不好的话整个组件看着会很混乱，不如选项式强制规定哪里是 method 哪里是 computed 哪里是 setup 来的清晰

**Reply by mark9804** [reply]
但是选项式切换到组合式是很快的，其实也不用纠结于这个

**Reply by BTMuli** [reply]
想问问你们有没有 CodeReview，今天发现 Github 是有这个功能的(*ﾟ∀ﾟ*)

**Reply by mark9804** [reply]
Review 是你提交 PR 之后由别人来审核的，别人审完给你个意见

<img width="673" alt="image" src="https://user-images.githubusercontent.com/9006264/199254436-26cb4cd2-f3a2-410b-b15e-a653e03395d8.png">

目前组里大概没有能担任 review 的重量级选手，基本上大家入行时间都差不多，review 交给 linter 还有 CodeQL 了

**Comment by mark9804** [comment]
昨晚灵活发布了个小模块

<img width="690" alt="image" src="https://user-images.githubusercontent.com/9006264/213993844-3abd15c5-7470-4429-aa31-ef0b80de349e.png">

**Comment by mark9804** [comment]
警钟敲烂！
<img width="480" alt="image" src="https://user-images.githubusercontent.com/9006264/233854584-bd7b5633-3619-4e8b-8d9a-6c7de05d89a8.png">

**Reply by BTMuli** [reply]
> 警钟敲烂！

身体素质没好到能996的程度，上午没课还想我起来？無駄だ( ᐛ )

**Comment by BTMuli** [comment]
忽然想起来蔚蓝已经开预约了，不知道什么时候开服，mark一下

**Reply by mark9804** [reply]
快跑，剧情看这样子已经和谐烂了，抛开世界观和剧情这游戏纯纯坐牢

**Reply by BTMuli** [reply]
> 快跑，剧情看这样子已经和谐烂了，抛开世界观和剧情这游戏纯纯坐牢

草，是这样？那我还是回星穹当个开服玩家好了
